clarity/contracts/arkadiko-auction-engine-v4-1.clar

```diff
@@ -0,0 +1,295 @@
+    ended-at: uint,
+  }
+)
+
+(define-read-only (get-auction-by-id (id uint))
+  (default-to
+    {
+      id: u0,
+      auction-type: "collateral",
+      collateral-amount: u0,
+      collateral-token: "",
+      collateral-address: (contract-call? .arkadiko-dao get-dao-owner),
+      vault-id: u0,
+      debt-to-raise: u0,
+      discount: u0,
+      total-collateral-sold: u0,
+      total-debt-burned: u0,
+      ended-at: u0,
+    }
+    (map-get? auctions { id: id })
+  )
+)
+
+;; @desc check if auction open (not enough debt raised + end block height not reached)
+;; @param auction-id; ID of the auction to be checked
+(define-read-only (get-auction-open (auction-id uint))
+  (let (
+    (auction (get-auction-by-id auction-id))
+  )
+    (if (>= (get total-debt-burned auction) (get debt-to-raise auction))
+      false
+      true
+    )
+  )
+)
+
+;; @desc toggles the killswitch of the auction engine
+(define-public (toggle-auction-engine-shutdown)
+  (begin
+    (asserts! (is-eq tx-sender (contract-call? .arkadiko-dao get-guardian-address)) (err ERR-NOT-AUTHORIZED))
+
+    (ok (var-set auction-engine-shutdown-activated (not (var-get auction-engine-shutdown-activated))))
+  )
+)
+
+;; @desc check if shutdown activated
+(define-read-only (shutdown-activated)
+  (or
+    (unwrap-panic (contract-call? .arkadiko-dao get-emergency-shutdown-activated))
+    (var-get auction-engine-shutdown-activated)
+  )
+)
+
+;; @desc mark vault as liquidated and start auction
+;; @param vault-id; the vault to liquidate
+;; @param vault-manager; vault manager
+;; @param coll-type; collateral types
+;; @param oracle; oracle
+;; @param ft; collateral token if not STX vault
+;; @param reserve; reserve to get collateral tokens from
+;; @param liquidation-pool; pool to get USDA from
+;; @param liquidation-rewards; contract to deposit rewards
+(define-public (start-auction
+  (vault-id uint)
+  (vault-manager <vault-manager-trait>)
+  (coll-type <collateral-types-trait>)
+  (oracle <oracle-trait>)
+  (ft <ft-trait>)
+  (reserve <vault-trait>)
+  (liquidation-pool <liquidation-pool-trait>)
+  (liquidation-rewards <liquidation-rewards-trait>)
+)
+  (let (
+    (vault (contract-call? .arkadiko-vault-data-v1-1 get-vault-by-id vault-id))
+  )
+    (asserts! (not (shutdown-activated)) (err ERR-EMERGENCY-SHUTDOWN-ACTIVATED))
+    (asserts! (is-eq (contract-of vault-manager) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "freddie"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of coll-type) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "collateral-types"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of oracle) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "oracle"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of reserve) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "sip10-reserve"))) (err ERR-NOT-AUTHORIZED))
```

Comment:
REQ: can't this be `stx-reserve` as well? If a vault is not stacking, we auction off STX straight away which is stored by stx-reserve